scipy/interpolate/_bsplines.py

```diff
@@ -404,21 +410,37 @@ def design_matrix(cls, x, t, k):
         x = _as_float_array(x, True)
         t = _as_float_array(t, True)
 
+        if extrapolate != 'periodic':
+            extrapolate = bool(extrapolate)
+
+        if k < 0:
+            raise ValueError("Spline order cannot be negative.")
         if t.ndim != 1 or np.any(t[1:] < t[:-1]):
             raise ValueError(f"Expect t to be a 1-D sorted array_like, but "
                              f"got t={t}.")
         # There are `nt - k - 1` basis elements in a BSpline built on the
         # vector of knots with length `nt`, so to have at least `k + 1` basis
-        # element we need to have at least `2 * k + 2` elements in the vector
+        # elements we need to have at least `2 * k + 2` elements in the vector
         # of knots.
         if len(t) < 2 * k + 2:
             raise ValueError(f"Length t is not enough for k={k}.")
-        # Checks from `find_interval` function
-        if (min(x) < t[k]) or (max(x) > t[t.shape[0] - k - 1]):
+        if not np.isfinite(t).all():
+            raise ValueError("Knots should not have nans or infs.")
```

Comment:
This check is already done in `_as_float_array` (lines 410-411), so i suggest removing it